Source/BugsnagPluginClient.m

```diff
@@ -0,0 +1,36 @@
+//
+//  BugsnagPluginClient.m
+//  Bugsnag
+//
+//  Created by Jamie Lynch on 12/03/2020.
+//  Copyright © 2020 Bugsnag. All rights reserved.
+//
+
+#import "BugsnagPluginClient.h"
+#import "BugsnagPlugin.h"
+#import "BugsnagLogger.h"
+
+@interface BugsnagPluginClient ()
+@property NSSet<id<BugsnagPlugin>> *plugins;
+@end
+
+@implementation BugsnagPluginClient
+
+- (instancetype _Nonnull)initWithPlugins:(NSMutableSet<id<BugsnagPlugin>> *_Nonnull)plugins {
+    if (self = [super init]) {
+        _plugins = [NSSet setWithSet:plugins];
+    }
+    return self;
+}
+
+- (void)loadPlugins {
+    for (id<BugsnagPlugin> plugin in self.plugins) {
+        @try {
+            [plugin load];
+        } @catch (NSException *exception) {
+            bsg_log_err(@"Failed to load plugin %@, continuing with initialisation.", plugin);
+        }
```

Comment:
This obviously only catches a subset of possible errors.  Are we happy with the compromise of loading plugins before the full set of Bugsnag error trapping is in place?

Reply:
After discussion attempted to reduce this compromise by loading the plugins at the very end of `[BugsnagClient start]`, and adding an additional test to verify that exceptions are caught when loading plugins.